.github/workflows/java_ci_with_maven_temurin.yml

```diff
@@ -0,0 +1,28 @@
+# This workflow will build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Java CI with Maven[temurin] 
```

Comment:
Why we need both `adopt` and `temurin`? Do we need to support others like `microsoft`?

refs: https://github.com/actions/setup-java#supported-distributions

Reply:
it's the default template of GitHub workflow, we can use it only for checks, not required.